backend/functions/src/api/resources/user-profile/user-profile.service.ts

```diff
@@ -0,0 +1,12 @@
+import { UserProfile } from '.';
+import { getUserProfileDocument, updateUserProfileDocument } from '../dal';
+
+export async function getUserProfileService(uid: string) {
+  const getProfile = await getUserProfileDocument(uid);
+  return getProfile;
+}
+
+export async function updateUserProfileService(uid: string, userProfile: UserProfile) {
+  const getProfile = await updateUserProfileDocument(uid, userProfile);
+  return getProfile;
+}
```

Comment:
You can simply return the `Promise` returned by the dal functions. Can also remove the `async` keyword.